packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts

```diff
@@ -1037,6 +1037,7 @@ export class PBRMetallicRoughnessBlock extends NodeMaterialBlock {
 
             #ifdef METALLICWORKFLOW
                 surfaceAlbedo = reflectivityOut.surfaceAlbedo;
+                vec4${state.fSuffix} vReflectivityColor = vec4${state.fSuffix}(${this.metallic.associatedVariableName}, ${this.roughness.associatedVariableName}, ${this.indexOfRefraction.associatedVariableName}, 1.0);

```

Comment:
Okay, I don't know what I'm doing here. I noticed that `vReflectivityColor` isn't defined for Node Materials.
I haven't used them before so I'm not familiar with how they're constructed. Am I doing something the wrong way?

Reply:
When using the regular PBR material (not node material), `vReflectivityColor` is a uniform passed to `reflectivityBlock()` (1st parameter).

With node materials, this uniform doesn't exist, that's why we directly pass a `vec4` instead of the variable (line 1018). So, you don't need line 1040, you can just remove it.

Reply:
Ah, I can see you are using `vReflectivityColor` in new places... So, you can fix the problems like this:

https://github.com/BabylonJS/Babylon.js/commit/d6cde39aa7d8dde5b689df076a3073bd875beb01

Reply:
Thank you. I'm not sure that I follow what's going on there though. What does it mean to exclude the variable name from the block?

Reply:
`state._excludeVariableName(“XXX”)` tells the system that a variable named “XXX” is declared and used directly by our code, so `_getFreeVariableName(“XXX”)` should never return “XXX” (it will return “XXX1”, “XXX2”, etc. instead).

We normally try not to use variables with hard-coded names and use `_getFreeVariableName(“XXX”)` to obtain a variable name with the name we want (“XXX”), but if this name is already registered, we will get “XXX1”, “XXX2”, etc. instead. But in some cases, the variable already exists in the code and it is easier to use this name, so we tell the system that this name is already in use to prevent someone wanting to create a variable with this name from getting “XXX” by calling _getFreeVariableName(“XXX”).

Reply:
(quick note - I will let you resolve this conversation when it is done, and will merge it after)